src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -1763,22 +1768,25 @@ internal static MethodBase GetMethodBase(RuntimeType reflectedType, RuntimeMetho
                     // Without this the reflectedType.Cache.GetMethod call below may return a MethodInfo
                     // object whose ReflectedType is string[] and DeclaringType is object[]. That would
                     // be (arguabally) incorrect because string[] is not a subclass of object[].
-                    MethodBase[] methodBases = reflectedType.GetMember(
+                    if (reflectedType.GetMember(
                         RuntimeMethodHandle.GetName(methodHandle), MemberTypes.Constructor | MemberTypes.Method,
-                        BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance) as MethodBase[];
-
-                    bool loaderAssuredCompatible = false;
-                    for (int i = 0; i < methodBases.Length; i++)
+                        BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+                        is MethodBase[] methodBases)
```

Comment:
Another "as" => "is" refactoring, maybe better just use explicit casting or "!" operator instead